XpringKit/JavaScript/JavaScriptUtils.swift

```diff
@@ -40,40 +45,62 @@ internal class JavaScriptUtils {
 		return result.toBool()
 	}
 
-    /// Encode the given classic address and tag into an x-address.
-    ///
-    /// - SeeAlso: https://xrpaddress.info/
-    ///
-    /// - Parameters:
-    ///   -  classicAddress: A classic address to encode.
-    ///   - tag: An optional tag to encode. Defaults to nil.
-    /// - Returns: A new x-address if inputs were valid, otherwise undefined.
-    public func encode(classicAddress: Address, tag: UInt32? = nil) -> Address? {
-      var arguments: [Any] = [ classicAddress ]
-      if tag != nil {
-        arguments.append(tag as Any)
-      }
+  /// Validate if the given string is a valid X-address on the XRP Ledger.
+  ///
+  /// - SeeAlso: https://xrpaddress.info/
+  ///
+  /// - Parameter address: An address to check.
+  /// - Returns true if the address is a valid X-address, otherwise false.
+  public func isValidXAddress(address: Address) -> Bool {
+    let result = isValidXAddressFunction.call(withArguments: [ address ])!
+    return result.toBool()
+  }
+
+  /// Validate if the given string is a valid  classic address on the XRP Ledger.
+  ///
+  /// - SeeAlso: https://xrpaddress.info/
+  ///
+  /// - Parameter address: An address to check.
+  /// - Returns true if the address is a valid classic address, otherwise false.
+  public func isValidClassicAddress(address: Address) -> Bool {
+    let result = isValidClassicAddressFunction.call(withArguments: [ address ])!
+    return result.toBool()
+  }
 
-      let result = encodeXAddressFunction.call(withArguments: arguments)!
-      return result.isUndefined ? nil : result.toString()
+  /// Encode the given classic address and tag into an x-address.
+  ///
+  /// - SeeAlso: https://xrpaddress.info/
+  ///
+  /// - Parameters:
+  ///   -  classicAddress: A classic address to encode.
+  ///   - tag: An optional tag to encode. Defaults to nil.
+  /// - Returns: A new x-address if inputs were valid, otherwise undefined.
```

Comment:
nit: usually we are capitalizing the X in X-address

Reply:
Done